ScanCommand/src/main/java/io/ballerina/scan/ProjectAnalyzer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan;
+
+import io.ballerina.projects.Document;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectKind;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * This class represents the project analyzer used for analyzing projects via core and external scanning methods.
+ *
+ * @since 0.1.0
+ * */
+class ProjectAnalyzer {
+    List<Issue> analyzeProject(Project project) {
+        List<Issue> allIssues = new ArrayList<>();
+        InternalScannerContext scannerContext = new InternalScannerContext(allIssues);
+
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            Module defaultModule = project.currentPackage().getDefaultModule();
+            defaultModule.documentIds().forEach(analyzeDocument(defaultModule, scannerContext));
+        } else {
```

Comment:
Won't it work if we use the same logic for Single file projects as well? Can you check if it will cause any errors or undesired outputs. If I'm not mistaken, it should work fine.
``` ballerina
project.currentPackage().moduleIds().forEach(moduleId -> { // there will be only one element here for single files
                Module module = project.currentPackage().module(moduleId);
                module.documentIds().forEach(analyzeDocument(module, scannerContext)); // 1 element for single files
                module.testDocumentIds().forEach(analyzeDocument(module, scannerContext)); // 0 elements for single files
            });
        }

Reply:
The suggested logic works